File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //Get text from user.
    string text = get_string("Text: ");

    //Calculating the number of letters.
    int letters = count_letters(text);

    //Calculating the number of words.
    int words = count_words(text);

    //Calculating the number of sentences.
    int sentences = count_sentences(text);

    float L = ((float) letters * 100) / (float) words;
    float S = ((float) sentences * 100) / (float) words;

    //Using the Coleman-Liau index.
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int index_int =  index;
    if ((index - index_int) > 0.5)
    {
        index_int++;
    }
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index_int);
    }

}

int count_letters(string text)
{
    int length = strlen(text);
    int count = 0;
    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int length = strlen(text);
    int count = 1;
    for (int i = 0; i < length; i++)
    {
        if (text[i] == ' ')
        {
            count++;
        }
    }
    return count;
}

int count_sentences(string text)
{
    int length = strlen(text);
    int count = 0;
    for (int i = 0; i < length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}

File: sentimental-readability/readability.py
from cs50 import get_string

# Gets text from user
text = get_string("Text: ")
words = 0
letters = 0
sentences = 0

# Calculates number os words, letters and sentences in the text
for i, items in enumerate(text):
    if (items.isspace()):
        words += 1
    if (items.isalpha()):
        letters += 1
    if (items == "!" or items == "?" or items == "."):
        sentences += 1
        if (i == len(text) - 1):
            words += 1

# Average number os letters and sentences per 100 words in the text
L = letters * 100 / words
S = sentences * 100 / words

# Coleman-Liau formula
formula = float(0.0588 * L - 0.296 * S - 15.8)
decimal = formula - int(formula)

# Rounds Coleman-Liau formula to the nearest integer
if (decimal > 0.50):
    formula = round(formula + 0.5)
else:
    formula = round(formula - 0.5)

# Prints Grade
if (formula < 1):
    print("Before Grade 1")
elif (formula >= 16):
    print("Grade 16+")
else:
    print(f"Grade {formula}")